packages/react/src/components/Masthead/Masthead.js

```diff
@@ -5,38 +5,173 @@
+import HeaderContainer from 'carbon-components-react/lib/components/UIShell/HeaderContainer';
+import UserProfile20 from '@carbon/icons-react/lib/user--profile/20';
+import { ReactComponent as Logo } from '../Icon/svg/ibm-logo.svg';
+import {
+  Header,
+  HeaderMenuButton,
+  HeaderGlobalBar,
+  HeaderGlobalAction,
+  HeaderName,
+  HeaderNavigation,
+  HeaderMenu,
+  HeaderMenuItem,
+  HeaderSideNavItems,
+  SkipToContent,
+  SideNav,
+  SideNavItems,
+  SideNavLink,
+  SideNavMenu,
+  SideNavMenuItem,
+} from 'carbon-components-react/lib/components/UIShell';
+import '@ibmdotcom/styles/scss/components/masthead/_masthead.scss';
+
+const { prefix } = settings;
 
 /**
- * Test MastHead component
+ * MastHead component
  *
+ * @typedef {object} navigation Object containing navigation elements
  * @param {string} type Type of masthead
  * @returns {*} Masthead component
- * @class
  */
-const Masthead = ({ type }) => {
-  if (type === 'branded') {
-    return (
-      <div className="masthead">
-        <MastheadSearch />
-      </div>
-    );
-  } else {
-    return (
-      <div className="masthead">
-        <MastheadSearch />
-      </div>
-    );
-  }
+const Masthead = ({ navigation }) => {
+  const navigationLinks = navigation.links;
+
+  return (
+    <HeaderContainer
+      render={({ isSideNavExpanded, onClickSideNavExpand }) => (
+        <>
+          <Header aria-label="IBM" data-autoid={`${prefix}--masthead`}>
+            <SkipToContent />
+            <HeaderMenuButton
+              aria-label="Open menu"
+              data-autoid={`${prefix}--masthead__hamburger`}
+              onClick={onClickSideNavExpand}
+              isActive={isSideNavExpanded}
+            />
+            {navigation.platform ? (
+              <>
+                <div
+                  className={`${prefix}--header__logo ${prefix}--header__logo--platform`}>
+                  <a href="https://ibm.com">
+                    <Logo data-autoid={`${prefix}--masthead__logo`} />
+                  </a>
+                </div>
+                <HeaderName
+                  prefix=""
+                  href={navigation.platform.path}
+                  data-autoid={`${prefix}--masthead__platform-name`}>
+                  {navigation.platform.name}
+                </HeaderName>
+              </>
+            ) : (
+              <div className={`${prefix}--header__logo`}>
+                <a href="https://ibm.com">
+                  <Logo data-autoid={`${prefix}--masthead__logo`} />
+                </a>
+              </div>
+            )}
+
+            <div className={`${prefix}--header__search`}>
+              <HeaderNavigation
+                aria-label="IBM"
+                data-autoid={`${prefix}--masthead__l0-nav`}>
+                {navigationLinks.map((item, index) => {
+                  if (item.subnav) {
+                    return (
+                      <HeaderMenu
+                        aria-label={item.name}
```

Comment:
Hm, not sure I follow. If there's a subnav it will render `<HeaderMenu />`, which will have the subnav items. Otherwise it renders an inline link `<HeaderMenuItem />`. Am I missing something?

Reply:
Again, are we talking about the same thing? I'm talking about `aria-label` here.

Reply:
Ah I get what you're saying. Yes, in your context it is redundant. I was just following how Carbon is using it. Let me take another look and see if I can get rid of one.